README.md

```diff
@@ -360,6 +360,7 @@ entities.
 | --- | --- | --- | --- | --- |
 | `type` | The ServiceNow CMDB configuration item type name | Y | `cmdb_ci_email_server` | |
 | `query` | An [encoded query string](https://docs.servicenow.com/csh?topicname=c_EncodedQueryStrings&version=utah&pubname=utah-platform-user-interface) to use to filter the result using  the `sysparm_query` parameter | N | `sys_updated_on>javascript:gs.dateGenerate('{{ .lastUpdateDate }}','{{ .lastUpdateTime }}')^operational_status!=2` | |
+| `extraQueryParms` | Additional Service Now [query params](https://docs.servicenow.com/bundle/vancouver-api-reference/page/integrate/inbound-rest/concept/c_TableAPI.html) other than `sysparm_query` parameter | N | `&sysparm_display_value=true&sysparm_exclude_reference_link=true` | |
```

Comment:
So, I understand that this is similar to what the old integration offered. However, I think we can do better with this implementation that makes it more apparent that we're using optional parameters. Instead of shoving the query params into a single string value, we could set an `optionalParams` map in the config like so:

```sh
- extEntityQuery:
    type: cmdb_ci_email_server
    query: 'sys_updated_on>javascript:gs.dateGenerate('${lastUpdateDate}','${lastUpdateTime}')^operational_status!=2'
    serverTimezone: America/Los_Angeles
    optionalParams:
      param1: value1
      param2: value2
      ...
```

These values should be read in and URI encoded appropriately. And then a helper func like `buildOptionalParams` can build the query param string and set them on the [same formatter which builds the url](https://github.com/newrelic/nr-entity-tag-sync/blob/9c2ba6f3b7db03189073ac53a82a0c004e15367b/internal/provider/servicenow/api.go#L112-L119).

Reply:
@jbeveland27  the 'extEntityQuery' is defined as map of strings in [config.go](https://github.com/newrelic/nr-entity-tag-sync/blob/fe73540f869382cc1b690f17127812888b748678/internal/sync/config.go#L26)